main.py

```diff
@@ -192,53 +206,89 @@ def add_workout_to_schedule(self,schedule,workout,duration):
         return self.post_req('schedulestep',data)
 
 
-    def delete_workout(self, id):
+    def delete_workout(self, id=None):
         workouts = self.get_req('workout')
         list_of_id = list()
         for workout in workouts['results']:
             list_of_id.append(workout['id'])
-
-        if id in list_of_id:
-            self.delete_req('workout',id)
-            return f"Workout with id {id} was deleted successfully "
+        if id is not None:
+            if id in list_of_id:
+                req = self.delete_req('workout',id)
+                return req, f"Workout with id {id} was deleted successfully "
+            else:
+                return False, f"Workout with id {id} was not found!"
         else:
-            return f"Workout with id {id} was not found!"
-
-
-
-    def delete_exercise(self,workout_id,day_id,exercise_id):
-        workouts = self.get_req('workout')
-        days = self.get_req('day')
-        exercises = self.get_req('set')
-        list_of_workout = list()
-        list_of_days = list()
-        list_of_exercises = list()
-
-        for workout in workouts["results"]:
-            list_of_workout.append(workout["id"])
-        if workout_id not in list_of_workout:
-            return f"The workout with id {workout_id} couldn't be found"
-
-        for day in days["results"]:
-            if day["training"] == workout_id:
-                list_of_days.append(day["id"])
-
-        if day_id not in list_of_days:
-            return f"The day in workout with id {workout_id} and id {day_id} doesn't exists"
-
-        for exercise in exercises["results"]:
-            if exercise["exerciseday"] == day_id:
-                list_of_exercises.append(exercise["id"])
+            if len(list_of_id) != 0:
+                for id1 in list_of_id:
+                    self.delete_req('workout',id1)
 
-        print(list_of_workout)
-        print(list_of_days)
-        print(list_of_exercises)
-
-        if exercise_id not in list_of_exercises:
-            return f"The exercise {exercise_id} in day {day_id} doesn't exists"
-
-        self.delete_req('set', exercise_id)
-        return "Exercise was deleted succesfully!!!!!"
+                return True
+            else:
+                return False, "There are no workouts to be deteled"
+
+
+
+    def delete_exercise(self,workout_id=None,day_id=None,exercise_id=None):
+        if (exercise_id is not None) and (day_id is not None) and (workout_id is not None):
```

Comment:
This works, but you could also look into the all() function present in Python for a nicer syntax.